docs/ENGINE.md

```diff
@@ -0,0 +1,82 @@
+        C -->  C[eventLoop];
+        C --> |input|D["execute()"]
+        D --> |input.PUUID|E[SessionStore]
+        E --> |"[]sessions"|D
+        D --> |"[]sessions"|F[Risk Engine]
+        F --> |"[]outcome"|D
+        D --> |"[]alert"|G[Alerting Relay]
+    end
+
+    subgraph AA["ETL Manager"]
+        Z["engineRelay"] --> |"invariant input"|C
+    end
+
+    subgraph AAA["Alerting Manager"]
+        G --> FF["eventLoop()"]
+    end
+```
+
+
+## Inter-Connectivity
+The ETL publishes `Invariant Input` to the Risk Engine using a relay channel. The Risk Engine will subscribe to this channel to receive and process updates as they are published by the ETL. The Risk Engine will also publish events to the Alerting system using a separate downstream relay channel. The Alerting system will subscribe to this channel to receive and process updates as they are published by the Risk Engine.
+
+## Invariant Session
+An invariant session refers the the execution and representation of a single invariant. An invariant session is uniquely identified by a `SUUID` and is associated with a single `PUUID`. An invariant session is created by the `EngineManager` when a user requests to run an active session. The `EngineManager` will create a new `InvariantSession` and pass it to the `RiskEngine` to be executed. The `RiskEngine` will then execute the invariant session and return an `InvalidationOutcome` to the `EngineManager`. The `EngineManager` will then create an `Alert` using the `InvalidationOutcome` and publish it to the Alerting system.
+
+
+## Session UUID (SUUID)
+The SUUID is a unique identifier that is used to identify a specific invariant session. The SUUID is generated by the `EngineManager` when a user requests to run a new invariant session. The SUUID is used to uniquely identify a specific invariant session. This allows the `EngineManager` to perform operations on a specific invariant session such as removing it or updating it. 
+
+A `SUUID` constitutes of a `UUID` and a `PID`. The `PUUID` is the ID of the invariant that the session is associated with. T
+
+A `SessionPID` is encoded using the following 3 byte array sequence:
+```
+            0           1           2           3
+            |-----------|-----------|-----------|
+               Network    Pipeline    Invariant
+                Type        Type        Type
+```
+
+
+## Invariant Input
+The invariant input is a struct that contains the following fields:
+* `PUUID` - The ID of the invariant that the input data is intended for
+* `Input` - Transit data that was generated by the ETL
+
+## Invariant
+An invariant is a logical execution module that defines some set of invalidation criteria. The invariant is responsible for processing the input data and determining if an invalidation has occured. If an invalidation has occured, the invariant will return a `InvalidationOutcome` that contains relevant metadata necessary for the `EngineManager` to create an `Alert`.
+
+### Addressing
```

Comment:
Are there any other comparable properties? Curious why this is the one that stands out

Reply:
[657a6d9](https://github.com/base-org/pessimism/pull/52/commits/657a6d927050e5efa341501dbd79f129582f1754)